integration-docs: Update for new format (Groove, Gosquared, Gogs, Gitea).

PieterCK opened the pull request:
As the changes were mostly to the docs format and not content, these are grouped together in a PR that updates 4 doc pages.
As part of https://github.com/zulip/zulip/issues/29592, update the following integration documentation pages:
**Screenshots and screen captures:**

<details>

<summary>

[Groove](https://zulip.com/integrations/doc/groove)
</summary>

![image](https://github.com/zulip/zulip/assets/108744694/d990f145-d218-4e2a-b862-293df272d3a6)
</details>

<details>

<summary>

[GoSquared](https://zulip.com/integrations/doc/gosquared)
</summary>

![image](https://github.com/zulip/zulip/assets/108744694/f0dd3d1f-dbf1-411e-a8df-b28d20ae46ae)
</details>

<details>
<summary>

[Gogs](https://zulip.com/integrations/doc/gogs)
</summary>

![image](https://github.com/zulip/zulip/assets/108744694/d65c4228-e389-4539-94bf-5e4dba43af36)
</details>

<details>
<summary>

[Gitea](https://zulip.com/integrations/doc/gitea)
</summary>

![image](https://github.com/zulip/zulip/assets/108744694/338b97b3-8bfb-41ef-97f0-22a7896bd427)
</details>

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

Niloth-p commented:
Feedback:
- Merging steps to reduce the number of steps
- Replacing macro {!git-webhook-url-with-branches.md!} with {!git-branches-additional-feature.md!}
- Adding the 'configuration options' section for the {!git-branches-additional-feature.md!}

PieterCK commented:
Updated docs

PieterCK commented:
@zulipbot add "mentor review" @sbansal1999, please review this too, thanks!

Niloth-p commented on zerver/webhooks/gosquared/doc.md:
```diff
@@ -1,23 +1,38 @@
-
-1. Set **Webhook URL** to the URL constructed above. Set **Name** to
-   a name of your choice, such as `Zulip`. Click on **Save Integration**.
+   on **Add new**. Set **Webhook URL** to the URL constructed above.
```
Also, we can use "generated" instead of "constructed" for the URL, across all the docs, to reflect the change that URLs are now generated.

PieterCK replied:
Good point!

Niloth-p replied:
There are more occurrences of the same, in the other docs :)

PieterCK commented:
Updated GoSquard doc

alya commented:
@Niloth-p does this look good to you now?

Niloth-p commented:
> @Niloth-p does this look good to you now?

I would say that the wording aspect has not yet been addressed in this PR, probably because it was not initially opened with that in mind.

Hi @PieterCK, during our discussion, when I raised the question about including phrasing updates, you mentioned your opinion that the formatting changes and the writing changes would be separate PRs/commits, so we decided to review just for the formatting and reordering changes. From the updates on #30726 before its merge, it's clear that the text rewrites would need to be incorporated as well. Could you please look into that?

You can refer to any of the recently merged integration docs PRs and use [this list](https://github.com/zulip/zulip/issues/29592#issuecomment-2210910796) I made as well. Feel free to add any more patterns that you notice, so that we can all refer to it throughout this batch of updates.

PieterCK commented:
Update: Added commit description for integrations with notable changes in their documentation. Replaced the word "constructed" to be "generated" instead. Re-worded and combined some steps to make the overall instructions more concise.

@Niloth-p Appreciate the thorough review Niloth! The [list](https://github.com/zulip/zulip/issues/29592#issuecomment-2210910796) is super helpful. Thank you :+1:

laurynmm commented:
@PieterCK - Thanks for working on this! I went ahead and did a maintainer review on these since it'd gone through the buddy review process. I pushed a few updates...

**Updates**:
- I simplified the Groove documentation so that it better follows our patterns.
- I split out the Go Squared instructions so that it matched the 1 step for navigation and 1 step for configuring format.
- I removed the if blocks around the event filtering macro.

**Updated screenshots**:

<details>
<summary>Groove</summary>

![Screenshot from 2024-07-08 22-03-37](https://github.com/zulip/zulip/assets/63245456/b9c79d11-048f-4195-9a57-766c4685bfb4)
</details>
<details>
<summary>GoSquared</summary>

![Screenshot from 2024-07-08 22-03-54](https://github.com/zulip/zulip/assets/63245456/8bd2ac4c-30dd-4078-a8b5-4e806d067850)
</details>
<details>
<summary>Gog</summary>

![Screenshot from 2024-07-08 22-04-10](https://github.com/zulip/zulip/assets/63245456/173656ba-710b-4cc0-ac5f-8e06e98a1a7e)
</details>
<details>
<summary>Gitea</summary>

![Screenshot from 2024-07-08 22-04-26](https://github.com/zulip/zulip/assets/63245456/97827195-18b1-4672-b603-f66fce362124)
</details>

alya commented:
Looking good! I noted on CZO that the Gitea screenshot could be better.

timabbott commented:
Merged, thanks @PieterCK and @laurynmm!
